lib/gollum-lib/page.rb

```diff
@@ -223,8 +223,12 @@ def toc_data
     #
     # Returns Hash of metadata.
     def metadata
-      formatted_data if markup.metadata == nil
-      markup.metadata || {}
+      unless @metadata
+        formatted_data if markup.metadata == nil
```

Comment:
Why is this statement within the `unless` block?

Reply:
I was thinking it needs to be executed only once, when `@metadata` is not yet initialized

Reply:
The call is just to ensure that the page's metadata has actually been parsed

Reply:
Remind me what `formatted_data` does?

Reply:
It returns the page's contents after being parsed through the filter chain (which is also where metadata, table of contents, etc. are being generated).

Reply:
It's to initialize `markup.metadata`. Prior to calling `markup.render`, the page's contents won't yet have been parsed, and so `markup.metadata` returns `nil`. So in constructing the page's metadata, we have to ensure that the underlying markup object has executed the filter chain.

(`formatted_data` itself stores the results in `@formatted_data`, so we don't end up rendering the contents multiple times)

Reply:
Got it. Maybe add a comment there to explain this?

Reply:
Yes, it is kind of counterintuitive 👍